packages/flutter_driver/lib/src/driver/vmservice_driver.dart

```diff
@@ -150,11 +149,31 @@ class VMServiceFlutterDriver extends FlutterDriver {
     }
 
     /// Waits for a signal from the VM service that the extension is registered.
-    /// Returns [_flutterExtensionMethodName]
-    Future<String> waitForServiceExtension() {
-      return isolate.onExtensionAdded.firstWhere((String extension) {
-        return extension == _flutterExtensionMethodName;
-      });
+    ///
+    /// Looks at the list of loaded extensions for the current [isolateRef], as
+    /// well as the stream of added extensions.
+    Future<void> waitForServiceExtension() async {
+      _log('Waiting for service extension');
+
+      final Future<bool> extensionAlreadyAdded = isolateRef
+        .loadRunnable()
+        .then((VMIsolate isolate) {
+          if (isolate.extensionRpcs.contains(_flutterExtensionMethodName)) {
+            return true;
+          }
+          // Never complete. Rely on the stream listener to find the service
+          // extension instead.
+          return Completer<bool>().future;
+        });
+      final Future<bool> extensionAddedEvent = isolate.onExtensionAdded
+        .contains(_flutterExtensionMethodName);
```

Comment:
Will this ever evaluate if the extension is already added? If not, we might need to cancel the stream subscription after `Future.any` returns.